### ISSUE TITLE
more nuanced unused variable warnings when destructuring an array

### ISSUE THREAD

**Issue by john-michael-murphy** [eslint/eslint#11262]
**What rule do you want to change?**
`no-unused-vars`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new default behavior

**Please provide some example code that this change will affect:**

```js
  function bar(...params) {
    if (foo) foo(...params, this);
    const [d, idx, array] = params;
    array.forEach(...)
    ...etc
  }
```

**What does the rule currently do for this code?**
Gives a warning for the first and second variables in the deconstructed array.

**What will the rule do after it's changed?**
Ignore the first and second variables.

**Are you willing to submit a pull request to implement this change?**
Yes

---
This is just a thought! One can avoid this warning with `const array = params[2]`, but that just doesn't seem very idiomatic.

**Comment by platinumazure** [comment]
Hi @john-michael-murphy, thanks for the issue.

This is working as designed: You can use blank values when destructuring arrays to signal that you don't intend to use those values.

```js
const [, , array] = params;
```

Maybe we could consider an enhancement, but I'm not personally convinced it's necessary since there are multiple ways around the issue.

**Comment by john-michael-murphy** [comment]
Ah! I didn't know that syntax! Thanks!
